### PR TITLE
feat: create mail service module

### LINKED ISSUE #7: Move email feature
move email feature implemented in [https://github.com/detrash/detrash-api-new/pull/6](https://github.com/detrash/detrash-api-new/pull/6/files) to [https://github.com/detrash/detrash-api](https://github.com/detrash/detrash-api)

### PR DESCRIPTION
**Pull request by yurimutti**
[Related](https://linear.app/detrash/issue/ENG-49/move-email-feature)

- create email service integration with resend
- create calculator module, service and controller
- add react-email to customize email templates

### CODE REVIEW

File: src/calculator/calculator.controller.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      subject: 'Contact Information',
+      react: CalculatorSupportEmail({
+        companyType: supportDto.companyType,
+        employeesQuantity: supportDto.employeesQuantity,
+        wasteFootPrint: supportDto.wasteFootPrint,
+      }),
+    };
+
+    try {
+      const { data, error } = await this.mailService.sendEmail(mail);
```
Comment (faustovanin) [inline src/calculator/calculator.controller.ts:43]: Allowing mail sending through the front-end might expose an attack vector, where attachers might use the front-end to dispatch mail messages. Is it possible to add some backend check on e-mail requests or even sending e-mails through the backend?

File: src/calculator/calculator.service.ts
Comment (faustovanin) [inline src/calculator/calculator.service.ts:1]: Are DTO logic items being developed in another branch? At first glance, it seems that the task is not complete yet.

File: .env.example [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+# Auth
 AUTH0_AUDIENCE=
 AUTH0_DOMAIN=
 
+# Cloud
 STORAGE_TYPE=
-
 AWS_DEFAULT_REGION=
 AWS_ACCESS_KEY_ID=
 AWS_SECRET_ACCESS_KEY=
```
Comment (faustovanin) [inline .env.example:12]: Is there any risk storing such variable on the front-end?

Comment (yurimutti) [reply .env.example:12]: But here is our backend application.

File: src/calculator/calculator.controller.ts (hunk 1 above)
Comment (yurimutti) [reply src/calculator/calculator.controller.ts:43]: Here is our backend @faustovanin.

File: src/calculator/calculator.service.ts
Comment (yurimutti) [reply src/calculator/calculator.service.ts:1]: The main goal on this feature is creating a wrapper to sent emails using or email provider (resend). 

After this, we need to input some information about Calculator feature on Database.  

But we're covering this pull request for email purposes.

**Review by faustovanin — APPROVED** [review]

**Pull request merged**